src/components/Profile/ProfileInputField.jsx

```diff
@@ -0,0 +1,98 @@
+// React Imports
+import React, { useState } from 'react';
+// Inrupt Imports
+import { useSession } from '@inrupt/solid-ui-react';
+// Material UI Imports
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import CheckIcon from '@mui/icons-material/Check';
+import ClearIcon from '@mui/icons-material/Clear';
+import EditIcon from '@mui/icons-material/Edit';
+import Input from '@mui/material/Input';
+import InputLabel from '@mui/material/InputLabel';
+// Utility Imports
+import { fetchProfileInfo, updateProfileInfo } from '../../model-helpers';
+
+/**
+ * messageFolderProps is an object that stores the props for the MessageFolder
+ * component
+ *
+ * @typedef profileInputFieldProps
+ * @type {object}
+ * @property {string} inputName - Name of input field
+ * @property {string} inputValue - Value of input field used for updating profile
+ * @property {(value: React.SetStateAction<null>) => void} setInputValue - Set
+ * function for inputValue
+ * @property {() => void} fetchProfileData - Handler function for fetching data
+ * from user's profile card
+ * @memberof typedefs
+ */
+
+/**
+ * ProfileInputField Component - Component that creates the editable inputs fields
+ * for the Profile page
+ *
+ * @memberof Inbox
+ * @name ProfileInputField
+ * @param {profileInputFieldProps} Props - Props used for NewMessage
+ * @returns {React.JSX.Element} React component for NewMessage
+ */
+const ProfileInputField = ({ inputName, inputValue, setInputValue, fetchProfileData }) => {
+  const { session } = useSession();
+  const [edit, setEdit] = useState(false);
+
+  const handleUpdateProfile = async (event) => {
+    event.preventDefault();
+    const profileData = await fetchProfileInfo(session);
+    const inputField = inputName === 'Name' ? 'profileName' : inputName.toLowerCase();
+
+    await updateProfileInfo(session, profileData, inputField, inputValue);
+
+    fetchProfileData();
+    setEdit(false);
+  };
+
+  const handleCancelEdit = () => {
+    fetchProfileData();
```

Comment:
I might be missing why we need to `fetchProfileData()` here. Seems to me that sets the name and organization, but there would be no reason to set those states as there was no change correct? This is canceling the edit.

Reply:
When editing the inputs, if you were to cancel after editing, it'll remain what the input was when you canceled it. This is because onChange uses a set function each time the state is updated.

For example, say input was "Apple" when you want to edit, but was "Appl" when you canceled it. It'll remain "Appl" rather than the original "Apple".